packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -21,27 +21,38 @@ @interface FVPFrameUpdater : NSObject
 @property(nonatomic, weak, readonly) NSObject<FlutterTextureRegistry> *registry;
 // The output that this updater is managing.
 @property(nonatomic, weak) AVPlayerItemVideoOutput *videoOutput;
-// The last time that has been validated as avaliable according to hasNewPixelBufferForItemTime:.
-@property(nonatomic, assign) CMTime lastKnownAvailableTime;
+@property(nonatomic) CVPixelBufferRef latestPixelBuffer;
+@property(nonatomic) dispatch_queue_t pixelBufferSynchronizationQueue;
 @end
 
 @implementation FVPFrameUpdater
 - (FVPFrameUpdater *)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry {
   NSAssert(self, @"super init cannot be nil");
   if (self == nil) return nil;
   _registry = registry;
-  _lastKnownAvailableTime = kCMTimeInvalid;
   return self;
 }
 
 - (void)displayLinkFired {
   // Only report a new frame if one is actually available.
   CMTime outputItemTime = [self.videoOutput itemTimeForHostTime:CACurrentMediaTime()];
   if ([self.videoOutput hasNewPixelBufferForItemTime:outputItemTime]) {
-    _lastKnownAvailableTime = outputItemTime;
+    dispatch_async(self.pixelBufferSynchronizationQueue, ^{
+      if (self.latestPixelBuffer) {
+        CFRelease(self.latestPixelBuffer);
+      }
+      self.latestPixelBuffer = [self.videoOutput copyPixelBufferForItemTime:outputItemTime
+                                                         itemTimeForDisplay:NULL];
+    });
     [_registry textureFrameAvailable:_textureId];
   }
 }
+
+- (void)dealloc {
+  if (_latestPixelBuffer) {
+    CFRelease(_latestPixelBuffer);
```

Comment:
Minor note: he releases of the buffer should use `CVBufferRelease`, and that doesn't need to be null-checked.